R/calculateEnrichment.R

```diff
@@ -266,7 +274,16 @@ addMedipsEnrichmentFactors <- function(qseaSet, BSgenome = NULL, exportPath = NU
   #                                    }, mc.cores = nCores) %>%
   #   do.call(rbind, . )
 
-  qseaSet@sampleTable <- qsea::getSampleTable(qseaSet) %>%
+  # enrichDataInput <- parallel::lapply(inputFileNames ,
+  #                                  function(x){
+  #                   calculateCpGEnrichment(x, BSgenome, exportPath = exportPath,
+  #                                      extend = extend, shift = shift, uniq = uniq,
+  #                                      chr.select = chr.select, paired = paired)
+  #                                    }, mc.cores = nCores) %>%
+  #   do.call(rbind, . )
+
+
+  qseaSet@sampleTable <- as.data.frame(qsea::getSampleTable(qseaSet)) %>%
```

Comment:
Ah, the problem is that `qsea` can't cope with the sampleTable being a tibble rather than a data.frame.

Can you move the `as.data.frame()` to the end of the pipe, rather than wrapping the two lines please.

It looks like this function is ignoring the non-enriched files (if they exist), that is commented out at the moment.

Reply:
I actually assume, that not `qsea` can't cope with the sampleTable being a tibble, but `dplyr::select` is the problem. Error message explicitly says that `dplyr::select` can't work on matrix (it's made for data frame). So either, change `dplyr::select` into another tool that can handle it, or declare the `sampleTable(qseaSet)` as data frame. But this has to be done, before it is piped into `dpylr::select`, that's why I choose to add it at this point. 
Or am I wrong?

Concerning the non-enriched files: I did not change anything in this part of the script. It was already commented out. But it would make sense to include it, as well, (as optional setting "if exists...").